### PR TITLE
add maskrcnn inference example

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

1. add maskrcnn inference example
2. handle non detection case

## How was this patch tested?

unit tests and example tests with coco 2017 valiadation dataset

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala [hunk 1]
```diff
@@ -196,36 +198,39 @@ class MaskRCNN(val inChannels: Int,
...
+
+      val boxNumber = boxesInImage(i)
+      if (boxNumber > 0) {
+        val maskPerImg = masks.narrow(1, start, boxNumber)
+        val bboxPerImg = bboxes[Tensor[Float]](i + 1)
+        val classPerImg = labels.narrow(1, start, boxNumber)
+        val scorePerImg = scores.narrow(1, start, boxNumber)
+
+        require(maskPerImg.size(1) == bboxPerImg.size(1),
+          s"mask number ${maskPerImg.size(1)} should be same with box number ${bboxPerImg.size(1)}")
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: `the same`  or `should be identical to`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala [hunk 2]
```diff
@@ -196,36 +198,39 @@ class MaskRCNN(val inChannels: Int,
...
+
+        // bbox resize to original size
+        if (height != originalHeight || width != originalWidth) {
+          BboxUtil.scaleBBox(bboxPerImg,
+            originalHeight.toFloat / height, originalWidth.toFloat / width)
+        }
+        // mask decode to original size
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: `decode mask` or `mask decoded`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+-f: the COCO image files location
+-m: the annotation json file location
+-o: generated seq files location
+-p: number of parallel
+
+## Data Processing
+Input data are transformed by several pipeline classes, such as ScaleResize, ChannelNormalize, ImageFeatureToBatch, etc.
+
+## Model
+You can download [preTrain MaskRCNN model](.....) for BigDL. This MaskRCNN model refers to [facebookresearch/maskrcnn-benchmark](https://github.com/facebookresearch/maskrcnn-benchmark), and the model backbone is **R-50-FPN**.
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/README.md]: Please add a link for this model

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Test.scala [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            ChannelNormalize(122.7717f, 115.9465f, 102.9801f) ->
+            MatToTensor[Float](),
+        toRGB = false
+      )
+      val evaluationSet = transformer(rddData)
+
+      val model = Module.loadModule[Float](param.model)
+
+      val result = model.evaluate(evaluationSet,
+        Array(MeanAveragePrecisionObjectDetection.createCOCO(81, topK = -1),
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Test.scala]: Why object detection not segmentation?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: ok, change to "decode mask"

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala (hunk 1 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala]: ok, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Test.scala (hunk 4 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/Test.scala]: ok, refactor to new MAP api.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala [hunk 5]
```diff
@@ -270,15 +275,14 @@ class RoiMTImageFeatureToBatch private[bigdl](width: Int, height: Int,
...
-    origSizeData(position) = img.getOriginalSize
   }
 
   override protected def createBatch(batchSize: Int): MiniBatch[Float] = {
     if (featureTensor == null) {
       featureTensor = Tensor(Storage[Float](featureData),
         storageOffset = 1, size = Array(batchSize, 3, height, width))
     }
-    RoiMiniBatch(featureTensor, labelData.view, isCrowdData.view, origSizeData.view)
+    RoiMiniBatch(featureTensor, labelData.view, isCrowdData.view, null)
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala]: No constructor with `null` as default ?

Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/MTImageFeatureToBatch.scala]: ok,changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/README.md (hunk 3 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/README.md]: ok, added

**Comment by zhangxiaoli73** [comment]
jenkins pass: http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/1000/

**Pull request merged**